### PR TITLE
[release-v1.40] Automated cherry pick of #5549: Fix race condition leading to `Pod`s without any projected token volumes

### PR DESCRIPTION
**Pull request by rfranzke**
/kind/bug
/area/security

Cherry pick of #5549 on release-v1.40.

#5549: Fix race condition leading to `Pod`s without any projected token volumes

**Release Notes:**
```bugfix user
A race condition has been fixed which can lead to pods without any projected token volumes for newly created shoots.
```

### CODE REVIEW

File: pkg/gardenlet/controller/shoot/shoot_control_reconcile.go [hunk 1]
```diff
@@ -315,6 +315,16 @@ func (r *shootReconciler) runReconcileShootFlow(ctx context.Context, o *operatio
 			Fn:           flow.TaskFn(botanist.Shoot.Components.ControlPlane.ResourceManager.Wait).SkipIf(o.Shoot.HibernationEnabled),
 			Dependencies: flow.NewTaskIDs(deployGardenerResourceManager),
 		})
+		deployShootNamespaces = g.Add(flow.Task{
+			Name:         "Deploying shoot namespaces system component",
```
Comment (ialidzhikov) [inline pkg/gardenlet/controller/shoot/shoot_control_reconcile.go:319]: There is 1 more task with the same name and it is not removed.

Comment (rfranzke) [reply pkg/gardenlet/controller/shoot/shoot_control_reconcile.go:319]: Sorry, what do you mean?

Comment (ialidzhikov) [reply pkg/gardenlet/controller/shoot/shoot_control_reconcile.go:319]: https://github.com/gardener/gardener/blob/2ce93bdeefa6993879018832780582ad417ccc1b/pkg/gardenlet/controller/shoot/shoot_control_reconcile.go#L418-L422

is not removed. Probably something went wrong during the rebase.

Comment (rfranzke) [reply pkg/gardenlet/controller/shoot/shoot_control_reconcile.go:319]: Ah, thank you! Again, nice catch! Thanks for your valuable feedback 🤝

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**